[easy] Fix macro redefined warnings in C code

After applying clang-format to the C code (#416, part of #393), we get these warnings:

```
In file included from lwt_unix_stubs.c:98:
./lwt_unix_unix.h:839:9: warning: 'GET_INET_ADDR' macro redefined [-Wmacro-redefined]
#define GET_INET_ADDR(v) (*((struct in_addr *)(v)))
        ^
/Users/antron/.opam/4.04-lwt/lib/ocaml/caml/socketaddr.h:51:9: note: previous definition is here
#define GET_INET_ADDR(v) (*((struct in_addr *) (v)))
        ^
In file included from lwt_unix_stubs.c:98:
./lwt_unix_unix.h:2365:9: warning: 'GET_INET6_ADDR' macro redefined [-Wmacro-redefined]
#define GET_INET6_ADDR(v) (*((struct in6_addr *)(v)))
        ^
/Users/antron/.opam/4.04-lwt/lib/ocaml/caml/socketaddr.h:55:9: note: previous definition is here
#define GET_INET6_ADDR(v) (*((struct in6_addr *) (v)))
        ^
2 warnings generated.
```

This is probably just because of [the space deleted by that PR](https://github.com/ocsigen/lwt/pull/416/files#diff-e7b477b8db74e783e0d540162677ec4fR839).

However, instead of reintroducing the space, I think we can delete these macro definitions. They were [added to Lwt](https://github.com/ocsigen/lwt/commit/d91e1f55f4539a1d4e8d58afe631b64b5b44c11d#diff-32dd5e49710c54ddd1d5c20fd2d11e79R2290) in 2010, and `caml/socketaddr.h` distributed with OCaml was [included](https://github.com/ocsigen/lwt/commit/155905ca7db15949801c30774042ec0ad573bfe9#diff-1ec0034c530cf0901cf4011f943aacdbR31) seven months ago, at which point the macros silently shadowed each other. We don't need the Lwt instances.

See [CONTRIBUTING.md](https://github.com/ocsigen/lwt/blob/master/doc/CONTRIBUTING.md#contributing-to-the-lwt-code) for the code workflow.

Other warnings:

```
lwt_unix_unix.h: In function ‘worker_readdir’:
lwt_unix_unix.h:1787:5: warning: ‘readdir_r’ is deprecated [-Wdeprecated-declarations]
     job->result = readdir_r(job->dir, job->entry, &job->ptr);
     ^~~
In file included from /usr/include/features.h:410:0,
                 from /usr/include/bits/libc-header-start.h:33,
                 from /usr/include/stdint.h:26,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/7.1.1/include/stdint.h:9,
                 from /usr/lib/ocaml/caml/config.h:99,
                 from /usr/lib/ocaml/caml/misc.h:24,
                 from /usr/lib/ocaml/caml/alloc.h:23,
                 from lwt_unix_stubs.c:29:
/usr/include/dirent.h:189:12: note: declared here
 extern int __REDIRECT (readdir_r,
            ^
In file included from lwt_unix_stubs.c:98:0:
lwt_unix_unix.h: In function ‘worker_readdir_n’:
lwt_unix_unix.h:1833:9: warning: ‘readdir_r’ is deprecated [-Wdeprecated-declarations]
         int result = readdir_r(job->dir, &job->entries[i], &ptr);
         ^~~
In file included from /usr/include/features.h:410:0,
                 from /usr/include/bits/libc-header-start.h:33,
                 from /usr/include/stdint.h:26,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/7.1.1/include/stdint.h:9,
                 from /usr/lib/ocaml/caml/config.h:99,
                 from /usr/lib/ocaml/caml/misc.h:24,
                 from /usr/lib/ocaml/caml/alloc.h:23,
                 from lwt_unix_stubs.c:29:
/usr/include/dirent.h:189:12: note: declared here
 extern int __REDIRECT (readdir_r,
            ^
```

```
File "_none_", line 1:
Warning 58: no cmx file was found in path for module Toploop, and its interface was not compiled with -opaque
```

--- a/src/unix/lwt_unix_unix.h
+++ b/src/unix/lwt_unix_unix.h
@@ -841,8 +841,6 @@ CAMLprim value lwt_unix_mcast_set_ttl(value fd, value ttl)
 #define VAL_MCAST_ACTION_ADD (Val_int(0))
 #define VAL_MCAST_ACTION_DROP (Val_int(1))
 
-#define GET_INET_ADDR(v) (*((struct in_addr *)(v)))
-
 CAMLprim value lwt_unix_mcast_modify_membership(value fd, value v_action,
                                                 value if_addr, value group_addr)
 {
@@ -2364,10 +2362,7 @@ struct job_gethostbyname {
 };
 
 CAMLexport value alloc_inet_addr(struct in_addr *inaddr);
-#define GET_INET_ADDR(v) (*((struct in_addr *)(v)))
-
 CAMLexport value alloc_inet6_addr(struct in6_addr *inaddr);
-#define GET_INET6_ADDR(v) (*((struct in6_addr *)(v)))
 
 static value alloc_one_addr(char const *a)
 {

